src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,352 @@
+          return [theme.spacings.small];
+        }
+        return [theme.spacings.small, theme.spacings.xxlarge];
+    }
+  },
+  iconSize({ size, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return 'small';
+        }
+        return 'xsmall';
+      case 'small':
+        if (!children) {
+          return 'medium';
+        }
+        return 'small';
+      case 'medium':
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+      case 'large':
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+      default:
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  borderRadius({ theme }) {
+    return theme.spacings.xxsmall;
+  },
+};
+
+const StyledButton = styled(TouchableHighlight)`
+  background-color: ${styles.backgroundColor};
+  border-radius: ${styles.borderRadius};
+  border: ${styles.border};
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+  testID,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const buttonColor = `${variantColor || 'primary'}`;
```

Comment:
`const color` it's already in the scope of button so no need to add that as a `buttonColor`